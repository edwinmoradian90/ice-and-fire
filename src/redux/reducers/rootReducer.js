import { combineReducers } from 'redux'
import mainReducer from '../reducers/mainReducer'
import categoriesReducer from '../reducers/categoriesReducer'

export default combineReducers({
    mainReducer,
    categoriesReducer
})