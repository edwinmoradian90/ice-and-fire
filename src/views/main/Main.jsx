import React, { useEffect } from 'react'
import axios from 'axios'
import { useDispatch, useSelector } from 'react-redux'
import { Container } from '../../utils/styledComponents/main'
import Header from '../../components/header/Header'
import MainList from '../../components/list/MainList'
import {
    FETCH_MAIN_REQUEST,
    FETCH_MAIN_SUCCESS,
    FETCH_MAIN_FAILURE
} from '../../redux/constants/constants'
import { Loader } from '../../components/loading/Loader'
const url = 'https://anapioficeandfire.com/api'


function Main() {
    const dispatch = useDispatch()
    const data = useSelector(state => state.mainReducer.data)
    const loading = useSelector(state => state.mainReducer.loading)
    useEffect(() => {
        dispatch({ type: FETCH_MAIN_REQUEST })
        axios.get(url)
            .then(res => {
                setTimeout(() => {
                    dispatch({
                        type: FETCH_MAIN_SUCCESS,
                        payload: res
                    })
                }, 1500)
            })
            .catch(err => {
                dispatch({
                    type: FETCH_MAIN_FAILURE,
                    payload: err
                })
            })
    }, [dispatch])

    return (
        <div>
            {
                !loading
                    ?
                    (

                        <>
                            <Header />
                            <Container
                                height='calc(100vh - 72px)'
                                justifyContent='flex-start'
                            >
                                <MainList
                                    data={data}
                                />
                            </Container>
                        </>
                    )
                    :
                    (
                        <Loader loading={loading} />
                    )
            }
        </div>
    )
}

export default Main