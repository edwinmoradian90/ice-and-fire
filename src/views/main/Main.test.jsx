import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { BrowserRouter as Router } from 'react-router-dom';
import rootReducer from '../../redux/reducers/rootReducer';
import Main from './Main';

describe('Main', () => {
    const store = createStore(rootReducer);
    it('matches snapshot', () => {
        const component = renderer.create(
            <Provider store={store}>
                <Router>
                    <Main />
                </Router>
            </Provider>
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});