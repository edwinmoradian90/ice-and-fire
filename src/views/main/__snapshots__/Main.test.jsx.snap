// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main matches snapshot 1`] = `
<div>
  <div
    className="header"
  >
    <a
      className="sc-AxjAm irMvbk"
      href="/"
      onClick={[Function]}
    >
      <h1
        className="sc-AxhCb kYEhcE appTitle"
      >
        Ice and Fire
      </h1>
    </a>
  </div>
  <div
    className="sc-AxirZ iPAaiT"
    height="calc(100vh - 72px)"
  />
</div>
`;
